script/tool/test/common/package_command_test.dart

```diff
@@ -4,53 +4,46 @@
 
 import 'package:args/command_runner.dart';
 import 'package:file/file.dart';
-import 'package:file/memory.dart';
 import 'package:flutter_plugin_tools/src/common/core.dart';
 import 'package:flutter_plugin_tools/src/common/package_command.dart';
 import 'package:git/git.dart';
 import 'package:mockito/annotations.dart';
-import 'package:mockito/mockito.dart';
 import 'package:test/test.dart';
 
 import '../mocks.dart';
 import '../util.dart';
-import 'package_command_test.mocks.dart';
 
 @GenerateMocks(<Type>[GitDir])
 void main() {
+  late GitDir gitDir;
   late RecordingProcessRunner processRunner;
+  late RecordingProcessRunner gitProcessRunner;
   late SamplePackageCommand command;
   late CommandRunner<void> runner;
-  late FileSystem fileSystem;
   late MockPlatform mockPlatform;
   late Directory packagesDir;
   late Directory thirdPartyPackagesDir;
 
+  SamplePackageCommand configureCommand({bool includeSubpackages = false}) {
```

Comment:
I extracted a helper like this in a couple of tests. Some of our unit tests have the unfortunate pattern of setting up a usual command in `setUp`, but then in a few specific tests creating a custom command instance that overrides more things, ignoring the one in `setUp`, and using the local one.

I didn't want to try to fix all of those in this PR since this is already quite large, but when I hit those cases I did pull out a helper so that changes like "add `gitDir: gitDir` because now the command needs that mocked out" can at least be made in only one place in the file instead of having to look at why a subset of a command's tests are failing and then updating them with exactly the same change.